cmd/query/app/server.go

```diff
@@ -328,5 +347,11 @@ func (s *Server) Close() error {
 		s.cmuxServer.Close()
 		errs = append(errs, s.conn.Close())
 	}
+	if s.httpConn != nil {
+		errs = append(errs, s.httpConn.Close())
+	}
+	if s.grpcConn != nil {
+		errs = append(errs, s.grpcConn.Close())
+	}
```

Comment:
are these necessary? My assumption was that http/grpc servers do close their listeners when you shutdown the server.

Reply:
I don't think they are. I just removed them and there are no routine leaks. I've also looked through the `Close` implementations of all three server types and it does look like they all do close the underlying listeners.

To be honest I just blindly copied these after our discussion here https://github.com/jaegertracing/jaeger/pull/5299#discussion_r1538300184, and didn't verify whether they were actually necessary.